roles/docker/tasks/main.yml

```diff
@@ -73,15 +73,19 @@
   retries: 4
   delay: "{{ retry_stagger | random + 3 }}"
   with_items: "{{ dockerproject_repo_key_info.repo_keys }}"
-  when: not (ansible_os_family in ["CoreOS", "Container Linux by CoreOS", "RedHat", "Suse"] or is_atomic)
+  when:
+    - not (ansible_os_family in ["CoreOS", "Container Linux by CoreOS", "RedHat", "Suse"] or is_atomic)
+    - use_docker_engine is defined and use_docker_engine
```

Comment:
you can use just 
`- use_docker_engine`
you have it in defaults with false, so it is defined